internal/hwmon/hwmon.go

```diff
@@ -202,25 +189,25 @@ func GetFans(chip gosensors.Chip) map[int]*fans.HwMonFan {
 				min = fans.MinPwmValue
 			}
 
-			label := getLabel(chip.Path, inputSubFeature.Name)
+			label := getLabel(chip.Path, feature.Name)
 
 			fan := &fans.HwMonFan{
 				Config: configuration.FanConfig{
 					ID:     label,
 					MinPwm: &min,
 					MaxPwm: &max,
 					HwMon: &configuration.HwMonFanConfig{
-						Index:     currentOutputIndex,
-						PwmOutput: pwmOutput,
-						RpmInput:  rpmInput,
+						Index:      channel,
+						PwmChannel: channel,
```

Comment:
If now there is an index as well as a channel, why do we need to change how the index is assigned?
Can we keep the numbering scheme of fan2go, while supporting the channels of hwmon is a mapping impossible?

Reply:
OK, so the idea here is that a "fan" in fan2go really is a "fan RPM sensor". Such a fan _may_ be controlled by a PWM, which is a _different_ sensor (from hwmon's point of view). It is _only by accident_ that often a given fan RPM sensor channel will match the PWM sensor channel for the PWM that controls that fan. Some PWMs control more than one fan (as is the case of the hardware I am using), some fans have no PWM, and sometimes the channels don't match. The kernel has _no idea_ about this mapping and it must be specified by the user in the general case. In my patch, for convenience, I assume and assign the fan RPM sensor channel as the PWM sensor channel.

Note that it is entirely possible for a fan RPM sensor and a PWM sensor controller that fan to be provided by different chips. My patch does not handle that. To fully handle all possible hardware out there, fan2go should probably have a PWM struct and a PWM -> FAN map. I don't have hardware that looks like that and it's a bigger change, so I punted on doing that.